generator/src/main/java/com/scottlogic/deg/generator/fieldspecs/DateTimeRestrictionsMergeOperation.java

```diff
@@ -1,27 +1,45 @@
 package com.scottlogic.deg.generator.fieldspecs;
 
+import com.google.inject.Inject;
 import com.scottlogic.deg.generator.constraints.atomic.IsOfTypeConstraint;
 import com.scottlogic.deg.generator.restrictions.*;
 
 import java.util.Optional;
 
 public class DateTimeRestrictionsMergeOperation implements RestrictionMergeOperation {
-    private static final DateTimeRestrictionsMerger dateTimeRestrictionsMerger = new DateTimeRestrictionsMerger();
+    private final DateTimeRestrictionsMerger merger;
+
+    @Inject
+    public DateTimeRestrictionsMergeOperation(DateTimeRestrictionsMerger merger) {
+        this.merger = merger;
+    }
 
     @Override
     public Optional<FieldSpec> applyMergeOperation(FieldSpec left, FieldSpec right, FieldSpec merging) {
-        DateTimeRestrictions dateTimeRestrictions = dateTimeRestrictionsMerger.merge(
+        MergeResult<DateTimeRestrictions> mergeResult = merger.merge(
             left.getDateTimeRestrictions(), right.getDateTimeRestrictions());
 
-        if (dateTimeRestrictions == null) {
-            return Optional.of(merging.withDateTimeRestrictions(
-                null,
-                FieldSpecSource.Empty));
+        if (!mergeResult.successful){
+            //no temporal values can be created
+
+            TypeRestrictions typeRestrictions = merging.getTypeRestrictions();
+            if (typeRestrictions == null){
+                typeRestrictions = DataTypeRestrictions.ALL_TYPES_PERMITTED;
+            }
+
+            return Optional.of(merging
+                .withTypeRestrictions(
+                    typeRestrictions.except(IsOfTypeConstraint.Types.TEMPORAL),
+                    left.getFieldSpecSource().combine(right.getFieldSpecSource())));
+        }
+
+        if (mergeResult.restrictions == null) {
+            return Optional.of(merging); //no change
```

Comment:
if this is no change then why is the `.withDateTimeRestrictions()` not present here?

Reply:
because there are no dateTime restrictions to add to the `FieldSpec`